packages/protocol/contracts/shared/tokenvault/ERC20Vault.sol

```diff
@@ -342,6 +399,46 @@ contract ERC20Vault is BaseVault {
         });
     }
 
+    /// @notice Lets a solver fulfil a bridging intent by transferring the bridged token amount
+    // to the recipient.
+    /// @param _op Parameters for the solve operation
+    function solve(SolverOp memory _op) external nonReentrant whenNotPaused {
+        // Verify that the required L2 batch containing the intent transaction has been proposed
+        bytes32 l2BlockMetaHash =
+            (ITaikoInbox(resolve(LibStrings.B_TAIKO, false)).getBlockV3(_op.l2BlockId)).metaHash;
+        require(l2BlockMetaHash == _op.l2BlockMetaHash, L2_METADATA_HASH_MISMATCH());
```

Comment:
The Bridge/Vaults are designed to be chain-agnostic, operating identically whether deployed on L1 or L2. However, including `l2BlockMetaHash` in `SolverOp` breaks this design principle.

When resolving `ITaikoInbox(resolve(LibStrings.B_TAIKO, false))`:
- On L1: Returns `TaikoInbox` contract
- On L2: Returns `TaikoAnchor` contract
- For the same rollup, this creates divergent behavior

I propose to remove `l2BlockMetaHash` from `SolverOp` and make `SolverOp` chain-agnostic to maintain architectural consistency.

Reply:
Our main intention is to allow solving for L2 -> L1 bridging where the solution transaction can be made conditional on a certain batch being included. This is particularly important when solvers solve on the basis of preconfirmed transactions. May I have some suggestions on how it would be nice to segregate or constrain this divergent behaviour to only be available on the L1 vault?

Reply:
I think we have at least two options:

1. Keep the code as-is, `getBlockV3` will fail as the resolved address on L2 is TaikoAnchor which doesn't have the `getBlockV3` function.
2.  Do something like this:

```
address taiko = resolve(LibStrings.B_TAIKO, false);
// the new ITaiko interface, which is implemented by both TaikoInbox and TaikoAnchor
require(ITaiko(taiko).isOnL1(), NOT_ON_LAYER_1());
bytes32 l2BlockMetaHash = ITaikoInbox(taiko).getBlockV3(_op.l2BlockId)).metaHash;
```